src/java.base/share/classes/sun/net/www/protocol/http/DigestAuthentication.java

```diff
@@ -530,8 +685,11 @@ private String computeDigest(
         "00000000", "0000000", "000000", "00000", "0000", "000", "00", "0"
     };
 
-    private String encode(String src, char[] passwd, MessageDigest md) {
-        md.update(src.getBytes(ISO_8859_1.INSTANCE));
+    private String encode(String src, char[] passwd, MessageDigest md, Charset charset) {
+        if (charset.equals(StandardCharsets.UTF_8)) {
+            src = Normalizer.normalize(src, Normalizer.Form.NFC);
+        }
+        md.update(src.getBytes(charset));
         if (passwd != null) {
```

Comment:
[RFC7616 ](https://datatracker.ietf.org/doc/html/rfc7616#section-4) says the password should also support unicode. It's OK we use the old code for compatibility reason.